### PR TITLE
docs: document how the garbage collector works

### PR DESCRIPTION
**Pull request by alamb**
# Rationale

There was some non trivial confusion (including by myself) with how the garbage collector works, and the effect of its various knobs. Documenting this would have helped significantly

Internal slack link: https://influxdata.slack.com/archives/CRK9M8L5Q/p1674503738608999?thread_ts=1674501201.787439&cid=CRK9M8L5Q

# Changes
Add docs that explain how the garbage collector works

### CODE REVIEW

File: docs/garbage_collector.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+* `INFLUXDB_IOX_GC_PARQUETFILE_CUTOFF`: this setting controls when
+  rows are deleted from the `parquet_file` table. Any row that has a
+  value of `to_delete` that is greater than this setting will be
+  deleted from the catalog.
+
+* `INFLUXDB_IOX_GC_OBJECTSTORE_CUTOFF`: this setting controls when
+  objects are actually deleted from the object store. Any object that
+  was created (according to the object store timestamp) longer than
+  this interval ago will be deleted.
```
Comment (lukebond) [inline docs/garbage_collector.md:43]: i would make a small tweak: `...longer than this interval ago and is not referenced in the catalog's parquet_file table will be deleted`

**Review by lukebond — APPROVED** [review]

File: docs/garbage_collector.md (hunk 1 above)
Comment (alamb) [inline docs/garbage_collector.md:43]:
```suggestion
  this interval ago and is not referenced in the catalog's `parquet_file` table
  will be deleted.
```

Comment (alamb) [reply docs/garbage_collector.md:43]:
```suggestion
  this interval ago and is not referenced in the catalog's `parquet_file` table
  will be deleted.
```

**Comment by alamb** [comment]
Thank you for the review @lukebond

**Pull request merged**